irods_specific_implementation/configuration.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef IRODS_REST_CPP_CONFIGURATION_HPP
+#define IRODS_REST_CPP_CONFIGURATION_HPP
+
+#include <nlohmann/json.hpp>
+
+#include <string>
+#include <string_view>
+
+namespace irods::rest::configuration
+{
+    /// \brief Returns the JSON object attached to the "rest_api" key in the JSON configuration.
+    ///
+    /// \throws json::exception If the "rest_api" key does not exist at the top of the JSON object.
+    auto rest_api() -> const nlohmann::json&;
```

Comment:
Naming help wanted on these function names

Reply:
internal - good for now.